tools/SourceKit/CMakeLists.txt

```diff
@@ -23,7 +24,16 @@ set(SOURCEKIT_BINARY_DIR ${SWIFT_BINARY_DIR})
 set(SOURCEKIT_RUNTIME_OUTPUT_INTDIR "${SWIFT_RUNTIME_OUTPUT_INTDIR}")
 set(SOURCEKIT_LIBRARY_OUTPUT_INTDIR "${SWIFT_LIBRARY_OUTPUT_INTDIR}")
 
-check_symbol_exists(dispatch_block_create "dispatch/dispatch.h" HAVE_DISPATCH_BLOCK_CREATE)
+check_include_files("xpc/xpc.h" HAVE_XPC_H_default)
+option(HAVE_XPC_H
+  "Whether or not the current SDK has xpc.h"
+  ${HAVE_XPC_H_default})
+
+check_symbol_exists(dispatch_block_create "dispatch/dispatch.h" HAVE_DISPATCH_BLOCK_CREATE_default)
```

Comment:
Ah, so part of the problem is that it _does_ respect `CMAKE_OSX_SYSROOT`, but Swift's CMake explicitly sets the `OSX` variables like so:

```
# Set the CMAKE_OSX_* variables in a way that minimizes conflicts.
if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin" AND NOT CMAKE_CROSSCOMPILING)
  set(CMAKE_OSX_SYSROOT "${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_PATH}")
  set(CMAKE_OSX_ARCHITECTURES "")
  set(CMAKE_OSX_DEPLOYMENT_TARGET "")
endif()
```

The problem is that `check_symbol_exists` and `check_include_files` ends up trying to build for the target SDK but against the host's architectures, which fails spectacularly when you e.g. are trying to compile for iOS from an x86_64 macOS host.

(@gottesmm suggested offline that I turn these into options because Swift's CMake doesn't set up cross-compilation properly -- curious to hear if he has any thoughts about these changes.)

Reply:
@gottesmm -- bumping this question, do you have any thoughts about this PR? I'd like to get these changes or new changes which obviate these sooner than later.